index.js

```diff
@@ -327,71 +318,131 @@ if(!helpers.isTestEnvironment()){
-
-    redis.addXeThruSensorData(req, res);
-    handleSensorRequest(locationid);
-});
-
-app.post('/api/doorTest', async(req, res) => {
-    const {locationid} = req.body;
-    await redis.addDoorTestSensorData(req, res);
-    await handleSensorRequest(locationid)
+app.post('/api/xethru', Validator.body(['locationid', 'state', 'rpm', 'distance', 'mov_f', 'mov_s']).exists(), async (req, res) => {
+    try {
+        const validationErrors = Validator.validationResult(req)
+
+        if(validationErrors.isEmpty()){
+            // eslint-disable-next-line no-unused-vars -- might be useful in the future to know what all we have access to in the body
+            const { deviceid, locationid, devicetype, state, rpm, distance, mov_f, mov_s } = req.body;
+
+            await redis.addXeThruSensorData(locationid, state, rpm, distance, mov_f, mov_s);
+            await handleSensorRequest(locationid);
+            res.status(200).send()
+        } else {
+            helpers.log(`Bad request, parameters missing ${JSON.stringify(validationErrors)}`)
+            res.status(400).send()
+        }
+    } catch (err) {
+        helpers.log(err)
+        res.status(500).send()
+    }
 })
 
-app.post('/api/door', async(request, response) => {
-    const coreId = request.body.coreid
-    const locationid = await db.getLocationIDFromParticleCoreID(coreId)
+app.post('/api/doorTest', Validator.body(['locationid', 'signal']).exists(), async(req, res) => {
+    try {
+        const validationErrors = Validator.validationResult(req)
 
-    if(!locationid){
-        helpers.log(`Error - no location matches the coreID ${coreId}`)
-        response.status(400).json('No location for CoreID')
-    }
-    else{
-        const message = JSON.parse(request.body.data)
-        const signal = message.data
-        var doorSignal
-        if (signal==IM21_DOOR_STATUS.OPEN){
-            doorSignal = SESSIONSTATE_DOOR.OPEN
-        }
-        else if (signal==IM21_DOOR_STATUS.CLOSED){
-            doorSignal = SESSIONSTATE_DOOR.CLOSED
-        }
-        else if (signal==IM21_DOOR_STATUS.LOW_BATT){
-            doorSignal = "LowBatt"
-        }
-        else if (signal==IM21_DOOR_STATUS.HEARTBEAT_OPEN || signal==IM21_DOOR_STATUS.HEARTBEAT_CLOSED){
-            doorSignal = "HeartBeat"
+        if(validationErrors.isEmpty()){
+            const { locationid, signal } = req.body
+
+            await redis.addDoorTestSensorData(locationid, signal)
+            await handleSensorRequest(locationid)
+            res.status(200).send()
+        } else {
+            helpers.log(`Bad request, parameters missing ${JSON.stringify(validationErrors)}`)
+            res.status(400).send()
         }
+    } catch (err) {
+        helpers.log(err)
+        res.status(500).send()
+    }
+})
+
+app.post('/api/door', Validator.body(['coreid', 'data.*.deviceid', 'data.*.data', 'data.*.control']).exists(), async(request, response) => {
```

Comment:
Where are `data.*.deviceid', 'data.*.data', 'data.*.control'` being used?

I see `request.data.data` on line 391, but don't see the others or the reason for the stars. Can you please help me understand?

Reply:
This came from a discussion with Mario about checking the content of data which is an object, this is the syntax for checking object keys within body for the express validator :)

Reply:
That makes sense for `request.data.data`, but I just don't see where we are using the other object keys.

Reply:
@mariocimet mentioned to look inside data for I think `deviceid` specifically so I checked what we were using in the tests to determine what a valid request would entail was my thinking

Reply:
@mariocimet , it doesn't look like our code would break if the door didn't send us the deviceid (or the others). What are you hoping to catch with this?

Reply:
Yeah now that it's mentioned isn't deviceid one of the columns we were wanting to remove too? 😅 I think it was more a point of consistency in any case re checking for things not explicitly mentioned in the route handlers

Reply:
I'd prefer the philosophy that we want to be as strict as we need to be and no stricter. If they want to change their API to return different values, we should only complain if they were values that we actually care about.

Reply:
I was just referring to the request.data.data bit which we actually use to tell us the door status.

Reply:
oh ok, so should I change it just to look for that one?

Reply:
My question more specifically too I think is that you can't look at `request.data` as far as I can see from the validator documentation, which is why I was looking for the data.data bit of `request.body`, is that incorrect?

"Creates a validation chain for one or more fields. They may be located in any of the following request objects:

req.body
req.cookies
req.headers
req.params
req.query"